server/service/src/processors/transfer/requisition/test.rs

```diff
@@ -1,23 +1,46 @@
 use chrono::NaiveDate;
 use repository::{
     mock::{insert_extra_mock_data, MockData, MockDataInserts},
-    EqualFilter, ItemRow, KeyValueStoreRow, KeyValueType, NameRow, RequisitionFilter,
+    ChangelogFilter, ChangelogRepository, ChangelogTableName, EqualFilter, ItemRow,
+    KeyValueStoreRepository, KeyValueStoreRow, KeyValueType, NameRow, RequisitionFilter,
     RequisitionLineFilter, RequisitionLineRepository, RequisitionLineRow, RequisitionRepository,
     RequisitionRow, RequisitionRowRepository, RequisitionRowStatus, RequisitionRowType,
     StorageConnection, StoreRow,
 };
 use util::{inline_edit, inline_init, uuid::uuid};
 
 use crate::{
-    processors::test_helpers::{delay_for_processor, exec_concurrent},
+    processors::test_helpers::{delay_for_processor, delayed_with_retries, exec_concurrent},
     requisition::{
         request_requisition::{UpdateRequestRequisition, UpdateRequestRequisitionStatus},
         response_requisition::{UpdateResponseRequisition, UpdateResponseRequisitionStatus},
     },
     service_provider::ServiceProvider,
+    sync::ActiveStoresOnSite,
     test_helpers::{setup_all_with_data_and_service_provider, ServiceTestContext},
 };
 
+pub(crate) fn requisition_processed(connection: &StorageConnection) -> bool {
+    let active_stores = ActiveStoresOnSite::get(connection).unwrap();
+    let changelog_repo = ChangelogRepository::new(connection);
+    let key_value_store_repo = KeyValueStoreRepository::new(connection);
+
+    let cursor = key_value_store_repo
```

Comment:
Sometime full processor operation would jump a few cursor, depending on the sync setup sometimes both stores are on the same site (processor unit test) and sometimes they are on different sites (integration test), when they are on the same site multiple operations will happen in the same processor task, each operation will update cursor (it's done this way to have the same logic for both use cases of sync configuration):

https://github.com/openmsupply/open-msupply/blob/caa3c882a5f11a047f18de213ffe7fea9aca4bea/server/service/src/processors/transfer/requisition/mod.rs#L86-L98

There [is a diagram](https://github.com/openmsupply/open-msupply/blob/develop/server/service/src/processors/transfer/requisition/doc/omSupply_requisition_transfer_same_site.png) for both stores on the same site (here requisition will be created, then linked in the same processor operation, jumping two change logs)

And extract from [this diagram](https://github.com/openmsupply/open-msupply/blob/develop/server/service/src/processors/transfer/doc/omSupply_transfer_processors.png) where stores are on different sites (in this case requisition will only be created on supply store site, jumping one change log, and linked when created requisition arrives at order store):

![Screenshot 2023-05-10 at 12 02 28 PM](https://github.com/openmsupply/open-msupply/assets/26194949/55f4e4a6-261b-4ba3-b58d-2799fc821c0d)